src/Symfony/Component/Form/Console/Descriptor/Descriptor.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Console\Descriptor;
+
+use Symfony\Component\Console\Descriptor\DescriptorInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Form\ResolvedFormTypeInterface;
+use Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ *
+ * @internal
+ */
+abstract class Descriptor implements DescriptorInterface
+{
+    /**
+     * @var SymfonyStyle
+     */
+    protected $output;
+    protected $type;
+    protected $ownOptions = array();
+    protected $overriddenOptions = array();
+    protected $inheritedOptions = array();
+    protected $extensionOptions = array();
+    protected $requiredOptions = array();
+    protected $parents = array();
+    protected $extensions = array();
+
+    /**
+     * {@inheritdoc}
+     */
+    public function describe(OutputInterface $output, $object, array $options = array())
+    {
+        $this->output = $output;
+
+        switch (true) {
+            case $object instanceof ResolvedFormTypeInterface:
+                $this->describeResolvedFormType($object, $options);
+                break;
+        }
+    }
+
+    abstract protected function describeResolvedFormType(ResolvedFormTypeInterface $resolvedFormType, array $options = array());
+
+    public function isRequiredOption($option)
+    {
+        return in_array($option, $this->requiredOptions, true);
+    }
+
+    public function hasParents()
+    {
+        return 0 !== count($this->parents);
+    }
+
+    public function getParents()
+    {
+        return array_keys($this->parents);
+    }
+
+    public function hasExtensions()
+    {
+        return 0 !== count($this->extensions);
+    }
+
+    public function getExtensions()
+    {
+        return array_keys($this->extensions);
+    }
+
+    public function collectOptions(ResolvedFormTypeInterface $type)
+    {
+        $this->parents = array();
+        $this->extensions = array();
+
+        $ownOptionsResolver = new OptionsResolver();
+        try {
+            $type->getInnerType()->configureOptions($ownOptionsResolver);
+        } catch (UndefinedOptionsException $e) {
```

Comment:
Could the following patch improve the situation, as it'll allow collecting all the defined options but might miss the overridden options instead?

<details>
<summary>See patch</summary>

```diff
diff --git a/src/Symfony/Component/Form/Console/Descriptor/Descriptor.php b/src/Symfony/Component/Form/Console/Descriptor/Descriptor.php
index 2c447ef670..6b04d4f8ee 100644
--- a/src/Symfony/Component/Form/Console/Descriptor/Descriptor.php
+++ b/src/Symfony/Component/Form/Console/Descriptor/Descriptor.php
@@ -86,20 +86,19 @@ abstract class Descriptor implements DescriptorInterface
         $this->parents = array();
         $this->extensions = array();
 
-        $ownOptionsResolver = new OptionsResolver();
-        try {
-            $type->getInnerType()->configureOptions($ownOptionsResolver);
-        } catch (UndefinedOptionsException $e) {
-        }
-
         if (null !== $type->getParent()) {
             $optionsResolver = clone $this->getParentOptionResolver($type->getParent());
         } else {
             $optionsResolver = new OptionsResolver();
         }
 
-        $this->ownOptions = array_diff($ownOptionsResolver->getDefinedOptions(), $optionsResolver->getDefinedOptions());
-        $this->overriddenOptions = array_intersect($ownOptionsResolver->getDefinedOptions(), $optionsResolver->getDefinedOptions());
+        $this->ownOptions = array_diff($type->getOptionsResolver()->getDefinedOptions(), $optionsResolver->getDefinedOptions());
+
+        try {
+            $type->getInnerType()->configureOptions($overriddenOptionsResolver = new OptionsResolver());
+            $this->overriddenOptions = array_intersect($overriddenOptionsResolver->getDefinedOptions(), $optionsResolver->getDefinedOptions());
+        } catch (UndefinedOptionsException $e) {
+        }
 
         $this->inheritedOptions = array();
         foreach ($this->parents as $class => $opts) {
``` 

</details>